libs/oidc/provider/src/lib/controllers/interaction/interaction.controller.ts

```diff
@@ -237,8 +205,58 @@ export class InteractionController {
   }
 
   @Post(':uid/confirm')
-  public confirmPost(@Param() params) {
+  public async confirmPost(@Param() params, @Req() req: Request, @Res() res: Response) {
     console.log(':uid/confirm', 'confirmPost', params);
+    try {
+      const interactionDetails = await this.providerService.provider.interactionDetails(req, res);
+      const {
+        prompt: { name, details },
+        params,
+        session: { accountId }
+      } = interactionDetails;
+      // assert.equal(name, 'consent');
+
+      let { grantId } = interactionDetails;
+      let grant;
+
+      if (grantId) {
+        // we'll be modifying existing grant in existing session
+        grant = await this.providerService.provider.Grant.find(grantId);
+      } else {
+        // we're establishing a new grant
+        grant = new this.providerService.provider.Grant({
+          accountId,
+          clientId: params.client_id as string
+        });
+      }
+
+      if (details.missingOIDCScope) {
+        grant.addOIDCScope((details.missingOIDCScope as string[]).join(' '));
+      }
+      if (details.missingOIDCClaims) {
+        grant.addOIDCClaims(details.missingOIDCClaims);
+      }
+      if (details.missingResourceScopes) {
+        // eslint-disable-next-line no-restricted-syntax
+        // TODO: Enable this again
```

Comment:
ESLint doesn't give me any issues. Might not actually be needed?

Can't tell what about that is supposed to be restricted syntax